### ISSUE TITLE
Improving neural networks by preventing co-adaptation of feature detectors

### ISSUE THREAD

**Issue by leemik3** [yskim0/One-Day-One-Paper#65]
## Improving neural networks by preventing co-adaptation of feature detectors

input과 output을 연결하는 관계가 복잡할 수록 overfitting 위험이 커진다. 
이 overfitting을 dropout을 사용하여 training data의 co-adaptation을 막음으로써 예방할 수 있다. 
(co-adaptation : feature detector가 특정 feature detector들과의 문맥에서만 의미가 있는 경우)
이 때, 0.5의 확률로 은닉층의 unit을 제거하여 unit이 다른 특정 unit들에 의존하지 않도록 한다. 
드롭아웃은 model averaging을 수행하는데 매우 효과적이다. 
test set error를 줄이는 좋은 방법은 다양한 네트워크들의 예측을 평균 내는 것이다. 
가장 표준의 방법은 별개의 네트워크들을 훈련시키고, 이 네트워크들을 test data로 적용한다. 하지만 이것은 연산량이 매우 많다. random dropout이 매우 많은 수의 서로 다른 네트워크들을 적당한 시간에 훈련할 수 있게 해준다. 
각각의 training case는 분명 다른 네트워크이지만,같은 가중치를 공유한다. 

### MNIST
작은 learning rate을 주고 가중치 제한 없이 pre-trained model에 적용하여 error 향상됨

### TIMIT (standard dataset used for evaluation
of automatic speech recognition systems)
input을 정규화, minibatch size 100, 3~5개의 layer과 각 layer의 2000~4000개 뉴런으로 학습, 성능 향상됨

### CIFAR-10
3 Convolution layers, 3 Max pooling layers, 1 locally connected layers
성능 향상됨

### ImageNet

5 Convolution layers, max pooling layers, 2 globally connected layer, 1000 way softmax layer, 가중치 제한 -> 성능 향상됨
